CAIPs/caip-x.md

```diff
@@ -0,0 +1,424 @@
+
+- **Trust:** This allows individuals to define their trusted peers, shaping their trust graph;
+- **Security Report:** This enables security expert to publish security insights about software components;
+- **Review:** This allows individuals to endorse or dispute any claims as well as any entity such as software components;
+- **Trust score:** This enables any trust computer to publish computed trust scores about software components and/or peers.
+
+This data which translates explicit trust signals, can be supplemented with more implicit on-chain and/or off-chain trust signals, such as `Proof of Humanity`, `Proof of Membership`, `Proof of Contributions`, `Proof of Attendences`, `Social Graphs`, and more.
+
+Software components can encompass any executable code, particularly those derived from decentralized ecosystems. This encompasses self-custodial wallets (like MetaMask), wallet extensions (such as Snaps), decentralized network clients (for instance, Geth), smart contracts, decentralized applications, among others.
+
+## Motivation
+<!--The motivation is critical for CAIP. It should clearly explain why the state of the art is inadequate to address the problem that the CAIP solves. CAIP submissions without sufficient motivation may be rejected outright.-->
+Software components within a decentralized web tend to be distributed permissionlessly. While this fosters permissionless innovation, it simultaneously exposes the system to potential vulnerabilities and scams.
+Most existing solutions for evaluating software components are centralized, necessitating trusted intermediaries. This reliance on intermediaries compromises the decentralized property of the ecosystem.
+By standardizing data to form a universally applicable trust graph reusable in any context, we strengthen the reliability of software components assessments powered by the communities.
+
+## Specification
+<!--The technical specification should describe the standard in detail. The specification should be detailed enough to allow competing, interoperable implementations. -->
+### Identification
+Decentralized Identifiers ([DID](https://www.w3.org/TR/did-core/)) or Content Identifier (CID) are used to identify subjects such as peers, software components as well as the claims / assertions themselves.
+They can also identify any issuers, which could be peers or a software entity like a trust computers.
+
+The standardized data framework specifies the use of the following identifiers:
+- **Peers:** `PKH` DID method (e.g. `did:pkh:eip155:1:<publicAddress>`, `did:pkh:bip122:<publicAddress>`, `did:pkh:solana:<publicAddress>`);
+- **Software components:** Custom identifiers for software components such as the checksum (e.g. `snap://<checksum>`, `did:pkh:eip155:1:<contractAddress>`);
+- **Assertions:** CID generated based on their contents according to [RFC 8785
+JSON Canonicalization Scheme (JCS)](https://www.rfc-editor.org/rfc/rfc8785);
+- **Software entities:** `KEY` or `PKH` DID method.
```

Comment:
Indeed, I will list them as options rather than specifications. However, do you suggest any other relevant identifiers for software components and for assertions?

Reply:
lots of package managers identify individual binaries by a hash, if so this seems to be a pretty neutral/generic way of URN-ifying hashes that works across some common hashes:
https://datatracker.ietf.org/doc/html/draft-thiemann-hash-urn-01

or multiformats?